packages/hub-extension/src/index.ts

```diff
@@ -0,0 +1,89 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) Jupyter Development Team.
+| Distributed under the terms of the Modified BSD License.
+|----------------------------------------------------------------------------*/
+
+import { Menu } from '@phosphor/widgets';
+
+import { ICommandPalette } from '@jupyterlab/apputils';
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { PageConfig, URLExt } from '@jupyterlab/coreutils';
+
+import { IMainMenu } from '@jupyterlab/mainmenu';
+
+/**
+ * The command IDs used by the plugin.
+ */
+export namespace CommandIDs {
+  export const controlPanel: string = 'hub:control-panel';
+
+  export const logout: string = 'hub:logout';
+}
+
+/**
+ * Activate the jupyterhub extension.
+ */
+function activateHubExtension(
+  app: JupyterFrontEnd,
+  palette: ICommandPalette,
+  mainMenu: IMainMenu
+): void {
+  const hubHost = PageConfig.getOption('hub_host');
+  const hubPrefix = PageConfig.getOption('hub_prefix');
+  const baseUrl = PageConfig.getBaseUrl();
+
+  // Bail if not running on JupyterHub.
+  if (!hubPrefix) {
+    return;
+  }
+
+  console.log('hub-extension: Found configuration ', {
+    hubHost: hubHost,
+    hubPrefix: hubPrefix
+  });
+
+  const category = 'Hub';
+  const { commands } = app;
+
+  commands.addCommand(CommandIDs.controlPanel, {
+    label: 'Control Panel',
+    caption: 'Open the Hub control panel in a new browser tab',
+    execute: () => {
+      window.open(hubHost + URLExt.join(hubPrefix, 'home'), '_blank');
+    }
+  });
+
+  commands.addCommand(CommandIDs.logout, {
+    label: 'Logout',
+    caption: 'Log out of the Hub',
+    execute: () => {
+      window.location.href = hubHost + URLExt.join(baseUrl, 'logout');
```

Comment:
As of #6087, the File menu also has a logout item. I think I would be in favor of these menu items being unified under the File menu (where users are accustomed to seeing quit-ish options) rather than a separate menu.

Reply:
Sounds good 👍